bot/kodiak/pull_request.py

```diff
@@ -283,13 +288,30 @@ async def merge(
                     )
                 else:
                     self.log.exception("failed to merge pull request", res=res)
+                if e.response is not None and e.response.status_code == 500:
+                    raise GitHubApiInternalServerError
                 # we raise an exception to retry this request.
                 raise ApiCallException("merge")
 
     async def queue_for_merge(self) -> Optional[int]:
         self.log.info("queue_for_merge")
         return await self.queue_for_merge_callback()
 
+    async def add_label(self, label: str) -> None:
+        """
+        add label to pull request
+        """
+        self.log.info("add_label", label=label)
+        async with self.client(
+            installation_id=self.install, owner=self.owner, repo=self.repo
+        ) as api_client:
+            res = await api_client.add_label(label, pull_number=self.number)
+            try:
+                res.raise_for_status()
+            except HTTPError:
+                self.log.exception("failed to delete label", label=label, res=res)
+                raise ApiCallException("add label")
```

Comment:
I realized while adding this endpoint that if we encounter an exception trying to add a label to a pull request we will not be able to disable the bot.

I'm not sure what to do about this.